olive/evaluator/olive_evaluator.py

```diff
@@ -415,81 +413,83 @@ def get_inference_settings(cls, metric: Metric, model: ONNXModelHandler) -> Dict
         if model_infrerence_settings:
             inference_settings.update(model_infrerence_settings)
 
-        metric_inference_settings = metric.get_inference_settings(cls.framework.lower())
+        metric_inference_settings = metric.get_inference_settings(Framework.ONNX.lower())
         if metric_inference_settings:
             inference_settings.update(metric_inference_settings)
 
         return inference_settings
 
-    def _inference(
-        self,
+
+class OnnxEvaluator(OliveEvaluator, OnnxEvaluatorMixin, framework=Framework.ONNX):
+    def __init__(self):
+        super().__init__()
+
+    @staticmethod
+    def get_session_wrapper(
         model: ONNXModelHandler,
         metric: Metric,
         dataloader: Dataset,
-        post_func=None,
-        device: Device = Device.CPU,
-        execution_providers: Union[str, List[str]] = None,
-    ) -> Tuple[OliveModelOutput, Any]:
+        device: Device,
+        execution_providers: List[str],
+    ) -> Tuple[OrtInferenceSession, Dict[str, Any]]:
+        """Get the session wrapper for the model."""
         # user.config.inference_settings > model.inference_settings > default inference_settings
-        inference_settings = self.get_inference_settings(metric, model)
+        inference_settings = OnnxEvaluator.get_inference_settings(metric, model)
         session = model.prepare_session(
             inference_settings=inference_settings,
             device=device,
             execution_providers=execution_providers,
         )
 
+        # prepare for io binding
+        io_config = model.get_io_config()
+        io_bind = OnnxEvaluator.io_bind_enabled(metric, model.inference_settings)
+        shared_kv_buffer = metric.user_config.shared_kv_buffer
+        use_fp16 = any(v == "float16" for v in io_config["input_types"])
```

Comment:
Let us say, we have pass flow as converter -> transformer_optimization_fp16, but not enforce input as float32. 
Will the model io_config get updated to float16? Or inherit the input model's io_config?
 Will that lead to wrong `use_fp16`?

Reply:
Even before, we were checking the input types from the input feed which we expect to be in the same types as the model's io config. So the original logic has the same issues I think. 
We only use `fp16` along with `shared_kv_buffer` for gqa like use cases so I think it can only work if the io is fp16 anyways?

IIRC, you added this case to mimic the logic in the ort llama example https://github.com/microsoft/onnxruntime/blob/913bdc7306e11b65644f733861684a3a460e8db0/onnxruntime/python/tools/transformers/models/llama/llama_inputs.py#L291? 
It used to check for fp16 but now uses `use_gqa`. Maybe we could update this according in the future. In this PR, I don't change any existing behavior or logic. Only checking io_config instead of input_feed since they should technically be the same.

Reply:
Make sense.